Directory.Build.targets

```diff
@@ -22,4 +22,20 @@
     <NETCoreAppMaximumVersion>$(MajorVersion).$(MinorVersion)</NETCoreAppMaximumVersion>
   </PropertyGroup>
 
+  <UsingTask TaskName="ValidateAssemblyInformationalVersion" AssemblyFile="$(InstallerTasksAssemblyPath)"/>
+  <Target Name="ValidateAssemblyInfoVersion"
+          AfterTargets="CopyFilesToOutputDirectory"
+          Condition="'$(ShouldValidateAssemblyInfoVersion)' == 'true'">
+
+    <PropertyGroup>
+      <_expectedVersion>$(ProductVersion)-$(VersionSuffix)</_expectedVersion>
+      <_expectedVersion Condition="'$(DotNetFinalVersionKind)' == 'release'">$(ProductVersion)</_expectedVersion>
```

Comment:
This feels more like we're testing the compiler at this point.  We could just as soon check further up the stack and examine that the property has the expected value.  At that point we are just adding redundancy by calculating the property value in two different ways.  🤔 

My initial ask for a test was more like: we have this branding value, and we have an API that's supposed to return the branding value, let's make sure they agree.